cppcache/test/mock/ClientMetadataMock.hpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#ifndef GEODE_CLIENTMETADATAMOCK_H_
+#define GEODE_CLIENTMETADATAMOCK_H_
+
+#include <gmock/gmock.h>
+
+#include "ClientMetadata.hpp"
+
+namespace apache {
+namespace geode {
+namespace client {
+class ClientMetadataMock : public ClientMetadata {
```

Comment:
I would rather you overload the ClientMetadata constructor to accept a bucket set than burden the system with yet more polymorphism so you can inject a bucket set.

Reply:
I am not sure I understand @mreddington, this is mock and this is how you mock classes using gmock.